### PR TITLE
Pass tenantId parameter to PhoenixRDD when reading tables

### PR DESCRIPTION
**Pull request by nico-pappagianis**
- Added corresponding tests
- Cleaned up some test code

### CODE REVIEW

File: phoenix-spark/src/it/resources/tenantSetup.sql [hunk 1]
```diff
@@ -15,3 +15,4 @@
 -- limitations under the License.
 
 CREATE VIEW IF NOT EXISTS TENANT_VIEW(TENANT_ONLY_COL VARCHAR) AS SELECT * FROM MULTITENANT_TEST_TABLE
+UPSERT INTO TENANT_VIEW (ORGANIZATION_ID, TENANT_ONLY_COL) VALUES ('defaultOrg', 'defaultData')
```
Comment (nico-pappagianis) [inline phoenix-spark/src/it/resources/tenantSetup.sql]: Adding some default data so we can know something was actually read in the "read" tests

File: phoenix-spark/src/it/scala/org/apache/phoenix/spark/PhoenixSparkITTenantSpecific.scala [hunk 2]
```diff
@@ -39,62 +56,78 @@ class PhoenixSparkITTenantSpecific extends AbstractPhoenixSparkIT {
       results.append((rs.getString(1), rs.getString(2)))
     }
     stmt.close()
-
-    results.toList shouldEqual DataSet
+    results.toList shouldEqual VerificationDataSet
   }
 
-  test("Can persist a dataframe using 'DataFrame.saveToPhoenix' on tenant-specific view") {
+  /*****************/
```
Comment (nico-pappagianis) [inline phoenix-spark/src/it/scala/org/apache/phoenix/spark/PhoenixSparkITTenantSpecific.scala]: Two tests for "read" functionality

File: phoenix-spark/src/it/scala/org/apache/phoenix/spark/PhoenixSparkITTenantSpecific.scala [hunk 3]
```diff
@@ -19,18 +19,35 @@ import org.apache.spark.sql.SQLContext
 import scala.collection.mutable.ListBuffer
```
Comment (nico-pappagianis) [inline phoenix-spark/src/it/scala/org/apache/phoenix/spark/PhoenixSparkITTenantSpecific.scala]: Added two tests under "Read Tests" 
General code cleanup

File: phoenix-spark/src/main/scala/org/apache/phoenix/spark/PhoenixRDD.scala [hunk 4]
```diff
@@ -18,21 +18,23 @@ import java.sql.DriverManager
...
 import org.apache.spark.sql.types._
+import org.apache.spark.sql.{DataFrame, Row, SQLContext}
 
 import scala.collection.JavaConverters._
 
 class PhoenixRDD(sc: SparkContext, table: String, columns: Seq[String],
-                 predicate: Option[String] = None, zkUrl: Option[String] = None,
+                 predicate: Option[String] = None,
+                 zkUrl: Option[String] = None,
+                 tenantId: Option[String] = None,
```
Comment (nico-pappagianis) [inline phoenix-spark/src/main/scala/org/apache/phoenix/spark/PhoenixRDD.scala]: PhoenixRDD now takes in a tenantId param that defaults to None

File: phoenix-spark/src/main/scala/org/apache/phoenix/spark/PhoenixRelation.scala [hunk 5]
```diff
@@ -42,8 +42,8 @@ case class PhoenixRelation(tableName: String, zkUrl: String, dateAsTimestamp: Bo
       requiredColumns,
       Some(buildFilter(filters)),
       Some(zkUrl),
-      new Configuration(),
-      dateAsTimestamp
+      conf = new Configuration(),
```
Comment (nico-pappagianis) [inline phoenix-spark/src/main/scala/org/apache/phoenix/spark/PhoenixRelation.scala]: I could have placed tenantId at the end of the PhoenixRDD class so we don't have to used the named parameters "conf" and "dateAsTimeStamp". I'll commit a change for that.

File: phoenix-spark/src/main/scala/org/apache/phoenix/spark/SparkContextFunctions.scala [hunk 6]
```diff
@@ -32,10 +32,10 @@ class SparkContextFunctions(@transient val sc: SparkContext) extends Serializabl
    */
 
   def phoenixTableAsRDD(table: String, columns: Seq[String], predicate: Option[String] = None,
-                        zkUrl: Option[String] = None, conf: Configuration = new Configuration())
+                        zkUrl: Option[String] = None, tenantId: Option[String] = None, conf: Configuration = new Configuration())
                         : RDD[Map[String, AnyRef]] = {
 
     // Create a PhoenixRDD, but only return the serializable 'result' map
-    new PhoenixRDD(sc, table, columns, predicate, zkUrl, conf).map(_.result)
+    new PhoenixRDD(sc, table, columns, predicate, zkUrl, tenantId, conf).map(_.result)
```
Comment (nico-pappagianis) [inline phoenix-spark/src/main/scala/org/apache/phoenix/spark/SparkContextFunctions.scala]: Now we pass tenantId to PhoenixRDD so we can read tenant-specific tables

File: phoenix-spark/src/main/scala/org/apache/phoenix/spark/SparkSqlContextFunctions.scala [hunk 7]
```diff
@@ -29,11 +29,13 @@ class SparkSqlContextFunctions(@transient val sqlContext: SQLContext) extends Se
...
   def phoenixTableAsDataFrame(table: String, columns: Seq[String],
-                               predicate: Option[String] = None, zkUrl: Option[String] = None,
+                               predicate: Option[String] = None,
+                               zkUrl: Option[String] = None,
+                               tenantId: Option[String] = None,
                                conf: Configuration = new Configuration): DataFrame = {
 
     // Create the PhoenixRDD and convert it to a DataFrame
-    new PhoenixRDD(sqlContext.sparkContext, table, columns, predicate, zkUrl, conf)
+    new PhoenixRDD(sqlContext.sparkContext, table, columns, predicate, zkUrl, tenantId, conf)
```
Comment (nico-pappagianis) [inline phoenix-spark/src/main/scala/org/apache/phoenix/spark/SparkSqlContextFunctions.scala]: Now we pass tenantId to PhoenixRDD so we can read tenant-specific tables as DataFrames

**Comment by stoty** [comment]
Already merged.

**Pull request closed without merging**